src/platform/plugins/shared/dashboard/public/dashboard_api/sections_manager.ts

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import fastIsEqual from 'fast-deep-equal';
+import { BehaviorSubject, Subject } from 'rxjs';
+import { v4 as uuidv4 } from 'uuid';
+
+import { i18n } from '@kbn/i18n';
+import { StateComparators } from '@kbn/presentation-publishing';
+
+import { DashboardSectionMap } from '../../common/dashboard_container/types';
+import { DashboardState } from './types';
+
+export function initializeSectionsManager(initialSections: DashboardSectionMap | undefined) {
+  const scrollToBottom$ = new Subject<void>();
+  const sections$ = new BehaviorSubject<DashboardSectionMap | undefined>(initialSections);
+  function setSections(sections?: DashboardSectionMap) {
+    if (!fastIsEqual(sections ?? {}, sections$.value ?? {})) sections$.next(sections);
+  }
+
+  return {
+    api: {
+      sections$,
+      addNewSection: () => {
+        const oldSections = sections$.getValue() ?? {};
+        const newId = uuidv4();
+        setSections({
+          ...oldSections,
+          [newId]: {
+            id: uuidv4(),
```

Comment:
Shouldn't this be `newId`?
```suggestion
            id: newId,
```

Reply:
Oooof!! That could have been nasty 🙈 Good catch!! Fixed in a0f7ed5c89b8183d34dd5f008ee27f9c71d8cdff